src/lib.rs

```diff
@@ -219,7 +219,14 @@ impl CspList {
             let violates = policy.does_request_violate_policy(request);
             if let Violates::Directive(directive) = violates {
                 let resource = ViolationResource::Url(request.url.clone());
-                violations.push(Violation { resource, directive });
+                violations.push(Violation {
+                    resource,
+                    directive: Directive {
+                        name: get_the_effective_directive_for_request(request).to_owned(),
```

Comment:
These were all missing step 1 of https://www.w3.org/TR/CSP/#create-violation-for-request